src/BenchmarkDotNet/Engines/Engine.cs

```diff
@@ -138,6 +146,9 @@ public Measurement RunIteration(IterationData data)
             if(!isOverhead)
                 IterationSetupAction();
 
+            // this line should be before GcCollect because some strategies can allocate managed memory
+            cacheClearingStrategy?.ClearCache(affinity);
```

Comment:
It doesn't make sense to run it for all kind of iterations (including Jitting; Overhead). Only workload iterations should be affected.

Reply:
I've changed it to:
```c#
            if (data.IterationMode == IterationMode.Workload && data.IterationStage == IterationStage.Actual)
            {
                // this line should be before GcCollect because some strategies can allocate managed memory
                cacheClearingStrategy?.ClearCache(affinity);
            }
```